From a44c042fe1ce04d188cb2f270577a038a2ebb9f1 Mon Sep 17 00:00:00 2001
From: Stephan Hartmann <stha09@googlemail.com>
Date: Sun, 22 Aug 2021 06:16:32 +0000
Subject: [PATCH] libstdc++: do not declare vector of const types

libstdc++ requires that value_type of std::vector is non-const.
---
 .../browser/performance_manager/mechanisms/page_discarder.cc  | 4 ++--
 1 file changed, 2 insertions(+), 2 deletions(-)

diff --git a/chrome/browser/performance_manager/mechanisms/page_discarder.cc b/chrome/browser/performance_manager/mechanisms/page_discarder.cc
index 427a22b..6cf96c2 100644
--- a/chrome/browser/performance_manager/mechanisms/page_discarder.cc
+++ b/chrome/browser/performance_manager/mechanisms/page_discarder.cc
@@ -29,7 +29,7 @@ namespace {
 // UrgentlyDiscardMultiplePages can keep reclaiming until the reclaim target is
 // met or there is no discardable page.
 bool DiscardPagesOnUIThread(
-    const std::vector<const WebContentsProxy>& proxies) {
+    const std::vector<WebContentsProxy>& proxies) {
   DCHECK_CURRENTLY_ON(content::BrowserThread::UI);
   bool result = false;
   for (auto proxy : proxies) {
@@ -55,7 +55,7 @@ bool DiscardPagesOnUIThread(
 void PageDiscarder::DiscardPageNodes(
     const std::vector<const PageNode*>& page_nodes,
     base::OnceCallback<void(bool)> post_discard_cb) {
-  std::vector<const WebContentsProxy> proxies;
+  std::vector<WebContentsProxy> proxies;
   proxies.reserve(page_nodes.size());
   for (auto* page_node : page_nodes) {
     proxies.push_back(page_node->GetContentsProxy());
-- 
2.31.1

